cmd/observer/node_builder/observer_builder.go

```diff
@@ -0,0 +1,915 @@
+	"github.com/onflow/flow-go/model/encodable"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/flow/filter"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/buffer"
+	finalizer "github.com/onflow/flow-go/module/finalizer/consensus"
+	"github.com/onflow/flow-go/module/id"
+	"github.com/onflow/flow-go/module/mempool/stdmap"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/module/synchronization"
+	"github.com/onflow/flow-go/network"
+	netcache "github.com/onflow/flow-go/network/cache"
+	cborcodec "github.com/onflow/flow-go/network/codec/cbor"
+	"github.com/onflow/flow-go/network/p2p"
+	"github.com/onflow/flow-go/network/validator"
+	"github.com/onflow/flow-go/state/protocol"
+	badgerState "github.com/onflow/flow-go/state/protocol/badger"
+	"github.com/onflow/flow-go/state/protocol/blocktimer"
+	storage "github.com/onflow/flow-go/storage/badger"
+)
+
+// ObserverBuilder extends cmd.NodeBuilder and declares additional functions needed to bootstrap an Access node
+// These functions are shared by staked and observer builders.
+// The Staked network allows the staked nodes to communicate among themselves, while the public network allows the
+// observers and an Access node to communicate.
+//
+//                                 public network                           staked network
+//  +------------------------+
+//  | observer 1             |<--------------------------|
+//  +------------------------+                           v
+//  +------------------------+                         +----------------------+              +------------------------+
+//  | observer 2             |<----------------------->| Access Node (staked) |<------------>| All other staked Nodes |
+//  +------------------------+                         +----------------------+              +------------------------+
+//  +------------------------+                           ^
+//  | observer 3             |<--------------------------|
+//  +------------------------+
+
+type ObserverBuilder interface {
+	cmd.NodeBuilder
+
+	// IsStaked returns True if this is an Access Node, False otherwise
+	IsStaked() bool
+}
+
+// ObserverServiceConfig defines all the user defined parameters required to bootstrap an access node
+// For a node running as a standalone process, the config fields will be populated from the command line params,
+// while for a node running as a library, the config fields are expected to be initialized by the caller.
+type ObserverServiceConfig struct {
+	staked                       bool // deprecated
+	bootstrapNodeAddresses       []string
+	bootstrapNodePublicKeys      []string
+	observerNetworkingKeyPath    string
+	bootstrapIdentities          flow.IdentityList // the identity list of bootstrap peers the node uses to discover other nodes
+	NetworkKey                   crypto.PrivateKey // the networking key passed in by the caller when being used as a library
```

Comment:
> Do you mean we do not need them in cmd/access?

I mean that once we separate ConsensusFollower from the observer, then we don't need this because the Observer is always run as a binary.

Which brings be back to my previous comment, which is that the Consensus Follower should not rely on code inside the `observer` module.

Reply:
Thank you for the review.
I added a candidate here: https://github.com/GetElastech/flow-go/commit/2cd2d49ccb2b94f6837b8977e382fa9cac889dba
What do you think?

Reply:
I also added the cmd/observer part refactored to follower/ in this change https://github.com/GetElastech/flow-go/commit/5f37270e3743c0058c1e0500c8e25d69fccee8ed

Reply:
Observer dependency removed from follower fixed: https://github.com/GetElastech/flow-go/commit/38267d0548637e39df732483c3df6369bc23221c